### PR TITLE
Add support for WalletConnect v2 SDK beta #4669

### LINKED ISSUE #4669: Upgrade (try out?) to WalletConnect v2 SDK beta
When it's available

### PR DESCRIPTION
**Pull request by oa-s**
Closes #4669

### CODE REVIEW

File: AlphaWallet/Core/Types/Session.swift [hunk 1]
```diff
@@ -41,6 +41,12 @@ class WalletSession {
     }
 }
 
+extension WalletSession: {
+    var capi10Account: CAIP10Account {
```
Comment (hboon) [inline AlphaWallet/Core/Types/Session.swift:45]: capi10Account -> caip10Account :)

**Review by hboon — COMMENTED** [review]
Doesn't build on Travis and locally after clearing cache. Is it a build caching issue?

Is forking the SDK no longer needed?

Only did a quick check, what is `WalletConnectV2Storage` for and why is it added (and not earlier)?

Any feedback for the SDK folks?

---

And for our own purposes, what do you think of using publishers to replace delegates, like this:

```
Sign.instance.sessionRequestPublisher
```

File: AlphaWallet/Core/Types/Session.swift (hunk 1 above)
Comment (oa-s) [reply AlphaWallet/Core/Types/Session.swift:45]: > WalletConnectV2Storage 

it is code parts that were in WalletConnectV2Provider, just extracted to separate storage.

Comment (oa-s) [reply AlphaWallet/Core/Types/Session.swift:45]: i guess the same for v1 is needed

Comment (oa-s) [reply AlphaWallet/Core/Types/Session.swift:45]: > And for our own purposes, what do you think of using publishers to replace delegates, like this:

i guess it's great feature, we can move forward to use the same approach somewhere. (but not really sure), all our code base is based on delegate pattern, is it good idea to refactor all of these?

**Comment by oa-s** [comment]
@hboon pr is ready for review, tested v2 works for me.

File: AlphaWallet/Core/Types/Session.swift (hunk 1 above)
Comment (hboon) [reply AlphaWallet/Core/Types/Session.swift:45]: Hold on to delegate for a while. I'm a bit worried it's harder to find (it's actually just 1 more place to look for, not so bad). At least it's not necessary yet, right?

**Review by hboon — APPROVED** [review]
Didn't test it. Merge it in then. Since it's not prod yet.

**Comment by hboon** [comment]
> @hboon can you please try to build it for you? it works for me, but not for Travis.

Oh, works for me now. Strange. I have cleared Travi's cache and trying to re-run it, but you can merge if you are ready.

**Pull request merged**